### ISSUE TITLE
feat: allow disabling rules via inline comments (e.g. <!-- gomarklint-disable -->)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#136]
## Overview

Add support for disabling gomarklint rules on specific lines or regions using HTML comment directives embedded in Markdown files. This is a core linter feature (not a lint rule itself) that lets authors suppress false positives without changing global configuration.

## Problem

Some Markdown files legitimately violate a rule in specific sections (e.g., generated content, vendor docs, intentional exceptions). Currently, there is no way to suppress individual violations inline — the only option is to disable the rule globally in `.gomarklint.yml`, which silences it across all files.

## Proposed directive syntax

| Directive | Scope |
|---|---|
| `<!-- gomarklint-disable -->` | Disable **all** rules from this line onward |
| `<!-- gomarklint-enable -->` | Re-enable **all** rules from this line onward |
| `<!-- gomarklint-disable rule-a rule-b -->` | Disable named rules from this line onward |
| `<!-- gomarklint-enable rule-a rule-b -->` | Re-enable named rules from this line onward |
| `<!-- gomarklint-disable-line -->` | Disable all rules for the **current line only** |
| `<!-- gomarklint-disable-line rule-a -->` | Disable named rules for the **current line only** |
| `<!-- gomarklint-disable-next-line -->` | Disable all rules for the **next line only** |
| `<!-- gomarklint-disable-next-line rule-a -->` | Disable named rules for the **next line only** |

## Detection logic

After collecting lint errors for a file, filter out any `LintError` whose line number falls within a disabled region or is tagged for single-line suppression.

### Examples

```markdown
<!-- gomarklint-disable no-bare-urls -->
https://example.com
<!-- gomarklint-enable no-bare-urls -->

Valid line here.

<!-- gomarklint-disable-next-line duplicate-heading -->
## Duplicate Heading
```

| Input | Directive | Result |
|---|---|---|
| bare URL on line 2 | `disable no-bare-urls` (line 1) + `enable` (line 3) | suppressed |
| duplicate heading on line 8 | `disable-next-line duplicate-heading` (line 7) | suppressed |
| violation outside any disabled region | — | reported |

## Implementation approach

1. **Parse directives** — scan `lines` before rule checks to build a `disabledRanges` structure:
   - `disableAll[lineN]` for block-level disable
   - `disabledRules[lineN]` for named rules
   - single-line and next-line variants expand to explicit line sets
2. **Filter errors** — in `collectErrors` (or a post-processing step), drop any `LintError` whose `Line` is covered by the computed disabled set.
3. No changes to individual rule checkers required.

## References

- eslint: https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments
- markdownlint: `<!-- markdownlint-disable -->` / `<!-- markdownlint-enable -->`
- Roadmap item: Extensibility → "Allow disabling specific rules via inline comments"

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to create

- `internal/linter/disable_comment.go` — parse disable/enable directives from lines; return a `disabledSet` (map of line → set of rule names, or "all")
- `internal/linter/disable_comment_test.go` — unit tests for directive parsing and range computation

### Files to modify

- `internal/linter/linter.go` — call directive parser in `collectErrors`; filter `allErrors` through the disabled set before returning
- `internal/linter/linter_test.go` — integration tests covering each directive variant

### Checklist

- [ ] Implement `ParseDisableComments(lines []string) DisabledSet` in `disable_comment.go`
  - Handle `gomarklint-disable` / `gomarklint-enable` (block, with/without rule names)
  - Handle `gomarklint-disable-line` / `gomarklint-disable-next-line` (single-line, with/without rule names)
- [ ] Implement error filtering in `collectErrors` using the parsed `DisabledSet`
- [ ] Unit tests in `disable_comment_test.go` (parsing + range expansion)
- [ ] Integration tests in `linter_test.go` (end-to-end suppression per directive)
- [ ] E2E test fixture covering inline disable directives (`make test-e2e`)
- [ ] `make test-all` passes
